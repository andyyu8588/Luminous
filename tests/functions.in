function add(a, b) {
    return a + b;
}

function div(a, b) {
    return a / b;
}

function printAdd(a, b) {
    print(add(a, b));
}

function recursion(a) {
    if(a equals 1) {
        return;
    }
    a = a - 1;
    print(a);
    recursion(a);
}

function fizzbuzz(num) {
    toPrint = "";
    if(num % 3 equals 0) {
        toPrint = toPrint + "fizz";
    }
    if(num % 5 equals 0) {
        toPrint = toPrint + "buzz";
    }
    print(toPrint);
}

function spam() {
    printAdd(1, 2);
    printAdd(3, 4);
    fizzbuzz(3);
    fizzbuzz(5);
    fizzbuzz(15);
    print(div(6, 3));
    print(div(3, 6));
    recursion(5);
}

spam();