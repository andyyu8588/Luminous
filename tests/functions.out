3
7
fizz
buzz
fizzbuzz
2
0.5
4
3
2
1
